src/osd/PeeringState.cc

```diff
@@ -403,17 +407,42 @@ void PeeringState::write_if_dirty(ObjectStore::Transaction& t)
   }
 }
 
+#ifndef WITH_SEASTAR
 void PeeringState::advance_map(
+#else
+seastar::future<> PeeringState::advance_map(
+#endif
   OSDMapRef osdmap, OSDMapRef lastmap,
   vector<int>& newup, int up_primary,
   vector<int>& newacting, int acting_primary,
   PeeringCtx &rctx)
 {
-  ceph_assert(lastmap == osdmap_ref);
+ceph_assert(lastmap == osdmap_ref);
+#ifndef WITH_SEASTAR
   psdout(10) << "handle_advance_map "
 	    << newup << "/" << newacting
 	    << " -- " << up_primary << "/" << acting_primary
 	    << dendl;
+#else
+  return seastar::do_with(std::move(newup), std::move(newacting),
```

Comment:
why not chaining these continuations?

Reply:
Hi, now, I've chained the continuations, but kept the do_with invocation, because I think with do_with we avoid copying newup and newacting, is this right?